### PR TITLE
Allow PPO to turn off advantage normalization

### PR DESCRIPTION
**Pull request by vwxyzjn**
## Description
Allow PPO to turn of advantage normalization.

## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) for new features and bug fixes) (See #762)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist:

- [ ] I've read the [CONTRIBUTION](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) guide (**required**)
- [x] I have updated the changelog accordingly (**required**).
- [x] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [x] I have updated the documentation accordingly.
- [x] I have reformatted the code using `make format` (**required**)
- [x] I have checked the codestyle using `make check-codestyle` and `make lint` (**required**)
- [x] I have ensured `make pytest` and `make type` both pass. (**required**)
- [x] I have checked that the documentation builds using `make doc` (**required**)

Note: You can run most of the checks using `make commit-checks`.

Note: we are using a maximum length of 127 characters per line

### CODE REVIEW

**Comment by Miffyli** [comment]
Thanks for the PR! Hmm, this seems like such an obvious oversight I wonder why we did not have a switch for it in the first place. I let @araffin to pitch on this before merging.

File: stable_baselines3/ppo/ppo.py [hunk 1]
```diff
@@ -146,6 +148,7 @@ def __init__(
         self.n_epochs = n_epochs
         self.clip_range = clip_range
         self.clip_range_vf = clip_range_vf
+        self.normalize_advantage = normalize_advantage
```
Comment (araffin) [inline stable_baselines3/ppo/ppo.py:151]: if I recall, there is a check above that is valid only if normalize advantage is True

**Review by araffin — COMMENTED** [review]
thanks for the PR =)
In fact, I wanted to add that feature in the past but forgot...

File: stable_baselines3/ppo/ppo.py (hunk 1 above)
Comment (vwxyzjn) [reply stable_baselines3/ppo/ppo.py:151]: Hey sorry I don't quite get this...

**Comment by vwxyzjn** [comment]
This CI failed because of some imports error `cannot import name 'GoalEnv' from 'gym' (/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/gym/__init__.py)`, which don't seem related to this PR.

**Comment by araffin** [comment]
> 

will be fixed in https://github.com/DLR-RM/stable-baselines3/pull/782

File: stable_baselines3/ppo/ppo.py (hunk 1 above)
Comment (araffin) [reply stable_baselines3/ppo/ppo.py:151]: read the lines above, there is a check that is no longer valid when `normalize_advantage=False` but this is minor

Comment (vwxyzjn) [reply stable_baselines3/ppo/ppo.py:151]: Are you referring to the lines here: https://github.com/DLR-RM/stable-baselines3/blob/3084b81182f956149fb9235237f1e78f028914f1/stable_baselines3/ppo/ppo.py#L125-L146? They do not seem to relate to `batch_size` and I don't see how `normalize_advantage=False` will make those checks invalid.

Comment (araffin) [reply stable_baselines3/ppo/ppo.py:151]: https://github.com/DLR-RM/stable-baselines3/issues/440, `batch_size=1` is ok when no normalization is used as mentioned in the comment ;)

Comment (araffin) [reply stable_baselines3/ppo/ppo.py:151]: but could you also add a "running" test, that just check we can run PPO with `normalize_advantage=False` ?

Comment (vwxyzjn) [reply stable_baselines3/ppo/ppo.py:151]: > https://github.com/DLR-RM/stable-baselines3/issues/440, batch_size=1 is ok when no normalization is used as mentioned in the comment ;)

Oh I see 😅, definitely did not think of that. Just want to make sure I understand, do you want to support `batch_size=1` if `normalize_advantage=False`?

Comment (araffin) [reply stable_baselines3/ppo/ppo.py:151]: yes if it is easy to have, no otherwise ;)

File: tests/test_run.py [hunk 2]
```diff
@@ -36,13 +36,15 @@ def test_a2c(env_id):
 
 @pytest.mark.parametrize("env_id", ["CartPole-v1", "Pendulum-v1"])
 @pytest.mark.parametrize("clip_range_vf", [None, 0.2, -0.2])
-def test_ppo(env_id, clip_range_vf):
+@pytest.mark.parametrize("normalize_advantage", [False, True])
```
Comment (araffin) [inline tests/test_run.py:39]: I will write a more simple test tomorrow for A2C and PPO, as this creates way too many tests (it doubles its number) for just toggling `normalize_advantage` ;)

**Review by araffin — APPROVED** [review]
LGTM, thanks =)
Could you do a similar PR to SB3 contrib? (to keep PPO implementations consistent)

**Comment by vwxyzjn** [comment]
Will do

**Comment by vwxyzjn** [comment]
Just want to make sure I understand. Are you suggesting adding the normalization flag to https://github.com/Stable-Baselines-Team/stable-baselines3-contrib/tree/master/sb3_contrib/ppo_mask?

**Comment by araffin** [comment]
> Just want to make sure I understand. Are you suggesting adding the normalization flag to https://github.com/Stable-Baselines-Team/stable-baselines3-contrib/tree/master/sb3_contrib/ppo_mask?

yep (and I will need to do that for the recurrent PPO branch)

**Pull request merged**